doc/blender.1
.TH "BLENDER" "1" "July 15, 2010" "Blender Blender 2\&.52 (sub 5) "

.SH NAME
blender \- a 3D modelling and rendering package
.SH SYNOPSIS
.B blender [args ...] [file] [args ...]
.br
.SH DESCRIPTION
.PP
.B blender
is a 3D modelling and rendering package. It is the in-house software of a high quality animation studio, Blender has proven to be an extremely fast and versatile design instrument. The software has a personal touch, offering a unique approach to the world of Three Dimensions.

Use Blender to create TV commercials, to make technical visualizations, business graphics, to do some morphing, or design user interfaces. You can easy build and manage complex environments. The renderer is versatile and extremely fast. All basic animation principles (curves & keys) are well implemented.

http://www.blender.org
.SH OPTIONS

Blender 2.52 (sub 5) Build
Usage: blender [args ...] [file] [args ...]
.br
.SS "Render Options:"

.TP
.B \-b or \-\-background <file>
.br
Load <file> in background (often used for UI\-less rendering)
.br

.TP
.B \-a or \-\-render\-anim
.br
Render frames from start to end (inclusive)
.br

.TP
.B \-S or \-\-scene <name>
.br
Set the active scene <name> for rendering
.br

.TP
.B \-f or \-\-render\-frame <frame>
.br
Render frame <frame> and save it.
.br
+<frame> start frame relative, \-<frame> end frame relative.
.br

.TP
.B \-s or \-\-frame\-start <frame>
.br
Set start to frame <frame> (use before the \-a argument)
.br

.TP
.B \-e or \-\-frame\-end <frame>
.br
Set end to frame <frame> (use before the \-a argument)
.br

.TP
.B \-j or \-\-frame\-jump <frames>
.br
Set number of frames to step forward after each rendered frame
.br

.TP
.B \-o or \-\-render\-output <path>
.br
Set the render path and file name.
.br
Use // at the start of the path to
.br
    render relative to the blend file.
.br
The # characters are replaced by the frame number, and used to define zero padding.
.br
    ani_##_test.png becomes ani_01_test.png
.br
    test\-######.png becomes test\-000001.png
.br
    When the filename does not contain #, The suffix #### is added to the filename
.br
The frame number will be added at the end of the filename.
.br
    eg: blender \-b foobar.blend \-o //render_ \-F PNG \-x 1 \-a
.br
    //render_ becomes //render_####, writing frames as //render_0001.png//
.br

.TP
.B \-E or \-\-engine <engine>
.br
Specify the render engine
.br
use \-E help to list available engines
.br

.IP

.SS "Format Options:"

.TP
.B \-F or \-\-render\-format <format>
.br
Set the render format, Valid options are...
.br
    TGA IRIS JPEG MOVIE IRIZ RAWTGA
.br
    AVIRAW AVIJPEG PNG BMP FRAMESERVER
.br
(formats that can be compiled into blender, not available on all systems)
.br
    HDR TIFF EXR MULTILAYER MPEG AVICODEC QUICKTIME CINEON DPX DDS
.br

.TP
.B \-x or \-\-use\-extension <bool>
.br
Set option to add the file extension to the end of the file
.br

.TP
.B \-t or \-\-threads <threads>
.br
Use amount of <threads> for rendering in background
.br
[1\-BLENDER_MAX_THREADS], 0 for systems processor count.
.br

.IP

.SS "Animation Playback Options:"

.TP
.B \-a <options> <file(s)>
.br
Playback <file(s)>, only operates this way when not running in background.
.br
    \-p <sx> <sy> Open with lower left corner at <sx>, <sy>
.br
    \-m Read from disk (Don't buffer)
.br
    \-f <fps> <fps\-base> Specify FPS to start with
.br
    \-j <frame> Set frame step to <frame>
.br

.IP

.SS "Window Options:"

.TP
.B \-w or \-\-window\-border
.br
Force opening with borders (default)
.br

.TP
.B \-W or \-\-window\-borderless
.br
Force opening with without borders
.br

.TP
.B \-p or \-\-window\-geometry <sx> <sy> <w> <h>
.br
Open with lower left corner at <sx>, <sy> and width and height as <w>, <h>
.br

.IP

.SS "Game Engine Specific Options:"

.TP
.B \-g Game Engine specific options
.br
\-g fixedtime Run on 50 hertz without dropping frames
.br
\-g vertexarrays Use Vertex Arrays for rendering (usually faster)
.br
\-g nomipmap No Texture Mipmapping
.br
\-g linearmipmap Linear Texture Mipmapping instead of Nearest (default)
.br

.IP

.SS "Misc Options:"

.TP
.B \-d or \-\-debug
.br
Turn debugging on
.br

.IP
* Prints every operator call and their arguments
.br
* Disables mouse grab (to interact with a debugger in some cases)
.br
* Keeps python sys.stdin rather then setting it to None
.br

.TP
.B \-\-debug\-fpe
.br
Enable floating point exceptions
.br

.IP

.TP
.B \-nojoystick
.br
Disable joystick support
.br

.TP
.B \-noglsl
.br
Disable GLSL shading
.br

.TP
.B \-noaudio
.br
Force sound system to None
.br

.TP
.B \-setaudio
.br
Force sound system to a specific device
.br
NULL SDL OPENAL JACK
.br

.IP

.TP
.B \-h or \-\-help
.br
Print this help text and exit
.br

.IP

.TP
.B \-y or \-\-enable\-autoexec
.br
Enable automatic python script execution (default)
.br

.TP
.B \-Y or \-\-disable\-autoexec
.br
Disable automatic python script execution (pydrivers, pyconstraints, pynodes)
.br

.IP

.TP
.B \-P or \-\-python <filename>
.br
Run the given Python script (filename or Blender Text)
.br

.TP
.B \-\-python\-console
.br
Run blender with an interactive console
.br

.TP
.B \-v or \-\-version
.br
Print Blender version and exit
.br

.TP
.B \-\-
.br
Ends option processing, following arguments passed unchanged. Access via python's sys.argv
.br

.SS "Other Options:"

.TP
.B /?
.br
Print this help text and exit (windows only)
.br

.TP
.B \-R
.br
Register .blend extension (windows only)
.br

.SS "Argument Parsing:"

    arguments must be separated by white space. eg
        "blender \-ba test.blend"
    ...will ignore the 'a'
        "blender \-b test.blend \-f8"
    ...will ignore 8 because there is no space between the \-f and the frame value
.br
.SS "Argument Order:"

Arguments are executed in the order they are given. eg
        "blender \-\-background test.blend \-\-render\-frame 1 \-\-render\-output /tmp"
    ...will not render to /tmp because '\-\-render\-frame 1' renders before the output path is set
        "blender \-\-background \-\-render\-output /tmp test.blend \-\-render\-frame 1"
    ...will not render to /tmp because loading the blend file overwrites the render output that was set
        "blender \-\-background test.blend \-\-render\-output /tmp \-\-render\-frame 1" works as expected.
.br
.br
.SH "ENVIRONMENT VARIABLES"
  \fIBLENDER_USER_CONFIG\fR Directory for user configuration files.
  \fIBLENDER_SYSTEM_CONFIG\fR Directory for system wide configuration files.
  \fIBLENDER_USER_SCRIPTS\fR Directory for user scripts.
  \fIBLENDER_SYSTEM_SCRIPTS\fR Directory for system wide scripts.
  \fIBLENDER_USER_DATAFILES\fR Directory for user data files (icons, translations, ..).
  \fIBLENDER_SYSTEM_DATAFILES\fR Directory for system wide data files.
  \fIBLENDER_SYSTEM_PYTHON\fR Directory for system python libraries.
  \fITMP\fR or \fITMPDIR\fR Store temporary files here.
  \fIPYTHONHOME\fR Path to the python directory, eg. /usr/lib/python.
.br
.br

.br
.SH SEE ALSO
.B yafaray(1)

.br
.SH AUTHORS
This manpage was written for a Debian GNU/Linux system by Daniel Mester
<mester@uni-bremen.de> and updated by Cyril Brulebois
<cyril.brulebois@enst-bretagne.fr> and Dan Eicher <dan@trollwerks.org>.


doc/blender.1.py
#!/usr/bin/python

# ##### BEGIN GPL LICENSE BLOCK #####
#
#  This program is free software; you can redistribute it and/or
#  modify it under the terms of the GNU General Public License
#  as published by the Free Software Foundation; either version 2
#  of the License, or (at your option) any later version.
#
#  This program is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  You should have received a copy of the GNU General Public License
#  along with this program; if not, write to the Free Software Foundation,
#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
#
# ##### END GPL LICENSE BLOCK #####

# <pep8 compliant>

import subprocess
import os

import time
import datetime


def man_format(data):
    data = data.replace("-", "\\-")
    data = data.replace("\t", "    ")
    # data = data.replace("$", "\\fI")

    data_ls = []
    for w in data.split():
        if w.startswith("$"):
            w = "\\fI" + w[1:] + "\\fR"

        data_ls.append(w)

    data = data[:len(data) - len(data.lstrip())] + " ".join(data_ls)

    return data


blender_bin = os.path.join(os.path.dirname(__file__), "../blender")

blender_help = subprocess.Popen([blender_bin, "--help"], stdout=subprocess.PIPE).communicate()[0].decode()

blender_version = subprocess.Popen([blender_bin, "--version"], stdout=subprocess.PIPE).communicate()[0].decode().strip()
blender_version = blender_version.split("Build")[0]

date_string = datetime.date.fromtimestamp(time.time()).strftime("%B %d, %Y")

filepath = os.path.splitext(__file__)[0]

file = open(filepath, "w")

fw = file.write

fw('.TH "BLENDER" "1" "%s" "Blender %s"\n' % (date_string, blender_version.replace(".", "\\&.")))

fw('''
.SH NAME
blender \- a 3D modelling and rendering package''')

fw('''
.SH SYNOPSIS
.B blender [args ...] [file] [args ...]''')

fw('''
.br
.SH DESCRIPTION
.PP
.B blender
is a 3D modelling and rendering package. It is the in-house software of a high quality animation studio, Blender has proven to be an extremely fast and versatile design instrument. The software has a personal touch, offering a unique approach to the world of Three Dimensions.

Use Blender to create TV commercials, to make technical visualizations, business graphics, to do some morphing, or design user interfaces. You can easy build and manage complex environments. The renderer is versatile and extremely fast. All basic animation principles (curves & keys) are well implemented.

http://www.blender.org''')

fw('''
.SH OPTIONS''')

fw("\n\n")

lines = [line.rstrip() for line in blender_help.split("\n")]

while lines:
    l = lines.pop(0)
    if l.startswith("Environment Variables:"):
        fw('.SH "ENVIRONMENT VARIABLES"\n')
    elif l.endswith(":"):  # one line
        fw('.SS "%s"\n\n' % l)
    elif l.startswith("-") or l.startswith("/"):  # can be multi line

        fw('.TP\n')
        fw('.B %s\n' % man_format(l))

        while lines:
            # line with no
            if lines[0].strip() and len(lines[0].lstrip()) == len(lines[0]):  # no white space
                break

            if not l:  # second blank line
                fw('.IP\n')
            else:
                fw('.br\n')

            l = lines.pop(0)
            l = l[1:]  # remove first whitespace (tab)

            fw('%s\n' % man_format(l))

    else:
        if not l.strip():
            fw('.br\n')
        else:
            fw('%s\n' % man_format(l))

# footer

fw('''
.br
.SH SEE ALSO
.B yafaray(1)

.br
.SH AUTHORS
This manpage was written for a Debian GNU/Linux system by Daniel Mester
<mester@uni-bremen.de> and updated by Cyril Brulebois
<cyril.brulebois@enst-bretagne.fr> and Dan Eicher <dan@trollwerks.org>.
''')

print("written:", filepath)
